Snapshot take and bake

jkowalcz opened the pull request:
Add in Getting and loading the latest snapshots to your validator.  

**I'm unsure of the syntax of a info block as preview does not accurately show them please review closely**

jkowalcz committed:
Snapshot take and bake

Add in Getting and loading the latest snapshots to your validator.  I'm unsure of the syntax of a info block as preview does not accurately show them please review closely

abhay commented on docs/mine-hnt/validators/troubleshooting.mdx:
````diff
@@ -20,24 +20,56 @@ This can be run directly on the Docker container as well:
+* Temporarily pause transaction sync via ```miner repair sync_pause```
+* Cancel any in-progress transaction sync via ```miner repair sync_cancel```
+* Load snapshot from downloaded file ```miner snapshot load /var/data/snap-<height>```
+* Resume transaction syncing via ```miner repair sync_resume```
````
nit: This shouldn't be necessary.

jkowalcz replied:
So I've has a few times where it's needed so I added in a line to check if it's active again, if not I documented how to resume it just in case.

abhay commented on docs/mine-hnt/validators/troubleshooting.mdx:
```diff
@@ -20,24 +20,56 @@ This can be run directly on the Docker container as well:
 * you just launched an AMI instance (if using AWS)
 
+## Grab the latest blessed snapshot from Helium
+* On a running Validator run `miner snapshot list`
```
I would also add a step to verify in the API on the heights available:

https://api.helium.io/v1/snapshots

jkowalcz replied:
OK good to know I didn't know that API endpoint.

jkowalcz replied:
I added the API endpoint in the docs.

abhay commented on docs/mine-hnt/validators/troubleshooting.mdx:
```diff
@@ -49,7 +81,7 @@ The most common cause of Perf/DKG penalties is bad luck (due to bad peers or oth
 A few other penalty root causes exist, each requiring detailed monitoring capabilities to troubleshoot.
 
-For the time/blocks when your validator accrued penalties, check your metrics dashboard (e.g., Grafana)
+For the time/blocks when your validator accrued penalties, check your metrics dashboard (e.g., Grafana) [Baselining Your Validator](https://docs.helium.com/mine-hnt/validators/requirements).
```
stray commit?

jkowalcz replied:
yes it was not clear, I added a sentence to the bottom of the section to point people to a "easier" way to look at your baseline performance.

abhay reviewed (changes requested):
Nice. Thanks for putting this in the docs.

kellybyrd commented on docs/mine-hnt/validators/troubleshooting.mdx:
````diff
@@ -20,24 +20,56 @@ This can be run directly on the Docker container as well:
+* Cancel any in-progress transaction sync via ```miner repair sync_cancel```
+* On the source Validator take a snapshot via `miner snapshot take /var/data/snap-<height>` using the height you determined from the first step. 
+* On the source Validator resume transaction syncing via ```miner repair sync_resume```
+* Move the `/var/data/snap-<height>` file into your root of the `validator_data` directory on the destination Validator
````
For docker builds, `/var/data/...` is inside the container, it will be `validator_data` on the host of the source validator.

jkowalcz replied:
So I added in "For this example `/var/data` is inside the docker container which is mapped to your `validator_data` directory on the host system" in the note section to make sure that is clear.

kellybyrd commented on docs/mine-hnt/validators/troubleshooting.mdx:
````diff
@@ -20,24 +20,56 @@ This can be run directly on the Docker container as well:
+* Move that file into your `validator_data` directory
+* Temporarily pause transaction sync via ```miner repair sync_pause```
+* Cancel any in-progress transaction sync via ```miner repair sync_cancel```
+* Load snapshot from downloaded file ```miner snapshot load /var/data/snap-<height>```
````
nit: This load cmd may "timeout" and look like it fails, but the load continues in the background. Same note applies to load from a trusted validator. Not sure if you need to mention this or not.

jkowalcz replied:
Added in " This command may look like it fails with a timeout error.  The load actually continues in the background and you will need to wait and review the blockchain height in a few minutes."

jkowalcz committed:
Update troubleshooting.mdx

Updated based on @kellybyrd and @abhay input.

jkowalcz committed:
Update troubleshooting.mdx

Updated List to include periods in bulleted lists consistently,  there are lists items that have more than once sentence so I included them on all lines.

abhay commented on docs/mine-hnt/validators/troubleshooting.mdx:
```diff
@@ -20,24 +20,57 @@ This can be run directly on the Docker container as well:
 
+## Grab the latest blessed snapshot from Helium
+* On a running Validator run `miner snapshot list` or you can use the API to get the latest blessed snapshot going to this web address `https://api.helium.io/v1/snapshots`.
+* Find the highest "Height" block and download the snapshot from Helium via `curl -O https://snapshots.helium.wtf/mainnet/snap-<height>`.
```
```suggestion
* Find the highest "Height" block and download the snapshot from Helium via `curl -O https://snapshots.helium.wtf/mainnet/snap-<height>`. If this snapshot is not available, try the next highest as the consensus group may have not agreed upon the snapshot.
```

abhay reviewed (dismissed):
one small addition but LGTM

kellybyrd commented:
LGTM

jkowalcz committed:
Update docs/mine-hnt/validators/troubleshooting.mdx

Including @abhay suggestion, late.

Co-authored-by: Abhay Kumar <abhay@opensynapse.net>

PharkMillups reviewed (approved):
Merging!
